### PR TITLE
MGMT-4079 set olm_operators during cluster creation

### PR DESCRIPTION
**Pull request by pkliczewski**
We need to set olm_operators during cluster creation to define which optional operators needs to be installed.

Signed-off-by: Piotr Kliczewski <piotr.kliczewski@gmail.com>

### CODE REVIEW

**Comment by pkliczewski** [comment]
@gamli75 @YuviGold please take a look, I tested this with one success test for cnv deployment

**Comment by gamli75** [comment]
@lalon4 can you please review this PR and make sure that it won't break OCS tests?

**Comment by gamli75** [comment]
@pkliczewski please hold and don't merge this PR until I'll approve it - we can't merge it now because QE will not be able to run tests on integration.
Please try to run one of the OCS tests with this code and see if it breaks the test.

**Comment by pkliczewski** [comment]
> Please try to run one of the OCS tests with this code and see if it breaks the test.

@gamli75 I know that OCS tests are broken and they need to be refactored. This change allows to run cnv e2e.

**Comment by pkliczewski** [comment]
| /test e2e-metal-assisted
-- | --

**Review by YuviGold — APPROVED** [review]
LGTM

**Comment by YuviGold** [comment]
/hold
@priyanka19-98 might found a bug

File: discovery-infra/test_infra/helper_classes/cluster.py
Comment (priyanka19-98) [reply discovery-infra/test_infra/helper_classes/cluster.py:137]: there is no field like `olm_operators` in the cluster definition.

File: discovery-infra/test_infra/helper_classes/cluster.py
Comment (pkliczewski) [reply discovery-infra/test_infra/helper_classes/cluster.py:140]: Correct, In my patch I am reverting changes I made related to cnv. The ocs tests are broken ATM and @machacekondra is working on fixing it.

File: discovery-infra/test_infra/helper_classes/cluster.py
Comment (pkliczewski) [reply discovery-infra/test_infra/helper_classes/cluster.py:137]: Correct

File: discovery-infra/test_infra/helper_classes/cluster.py
Comment (machacekondra) [reply discovery-infra/test_infra/helper_classes/cluster.py:140]: Updated my PR with working solution: https://github.com/openshift/assisted-test-infra/pull/579

File: discovery-infra/test_infra/helper_classes/cluster.py
Comment (eranco74) [inline discovery-infra/test_infra/helper_classes/cluster.py:135]:
```suggestion
        logging.info(Setting Ocs to:{ocs_enabled} for cluster: {self.id}')
```

File: discovery-infra/tests/base_test.py
Comment (eranco74) [inline discovery-infra/tests/base_test.py:67]:
```suggestion
                             olm_operators=env_variables.get('olm_operators', None):
```

File: discovery-infra/test_infra/consts.py
Comment (eranco74) [inline discovery-infra/test_infra/consts.py:16]: Where is this used?

**Review by eranco74 — COMMENTED** [review]
Where is the code that verifies the olm_operators status at the end of the installation?

File: discovery-infra/test_infra/helper_classes/cluster.py
Comment (pkliczewski) [reply discovery-infra/test_infra/helper_classes/cluster.py:135]: The code is in downstream repo. I apply your suggestion but I need to make correct string. You forgot `'`

File: discovery-infra/tests/base_test.py
Comment (pkliczewski) [reply discovery-infra/tests/base_test.py:67]: We have default value for this variable set [here](https://github.com/openshift/assisted-test-infra/pull/601/files#diff-fd319f80628e389daf226c94bb6aa50ae1fce37b80a4b7221f6da81fbe0094f8R65)

File: discovery-infra/test_infra/consts.py
Comment (pkliczewski) [reply discovery-infra/test_infra/consts.py:16]: I want to use it as a timeout to wait on cnv related validations. Used in downstream repo.

**Comment by YuviGold** [comment]
Needs rebase after https://github.com/openshift/assisted-test-infra/pull/579 been merged.
/lgtm cancel

**Comment by pkliczewski** [comment]
@YuviGold please take a look

**Comment by YuviGold** [comment]
LGTM

**Pull request merged**
